File: adapters/src/main/java/adapters/dynamo/DynamoCustomer.java
package adapters.dynamo;

import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;

@DynamoDbBean
public class DynamoCustomer {
    private String ID;
    private String username;
    private DynamoCustomerData dynamoCustomerData;

    @DynamoDbPartitionKey
    public String getID() {
        return ID;
    }

    public void setID(String ID) {
        this.ID = ID;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public DynamoCustomerData getDynamoCustomerData() {
        return dynamoCustomerData;
    }

    public void setDynamoCustomerData(DynamoCustomerData dynamoCustomerData) {
        this.dynamoCustomerData = dynamoCustomerData;
    }
}

File: adapters/src/main/kotlin/adapters/customers/dynamo/DynamoCustomer.kt
package adapters.customers.dynamo

import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSecondaryPartitionKey

@DynamoDbBean
class DynamoCustomer {
    @get:DynamoDbPartitionKey
    var ID: String? = null
    @get:DynamoDbSecondaryPartitionKey(indexNames = ["UsernameIndex"])
    var username: String? = null
    var dynamoCustomerData: DynamoCustomerData? = null
}
